homeassistant/components/cover/__init__.py

```diff
@@ -0,0 +1,234 @@
+    hass.services.call(DOMAIN, SERVICE_CLOSE_COVER_TILT, data)
+
+
+def set_cover_tilt_position(hass, tilt_position, entity_id=None):
+    """Move to specific tilt position all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    data[ATTR_TILT_POSITION] = tilt_position
+    hass.services.call(DOMAIN, SERVICE_SET_COVER_TILT_POSITION, data)
+
+
+def stop_cover_tilt(hass, entity_id=None):
+    """Stop all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_STOP_COVER_TILT, data)
+
+
+def setup(hass, config):
+    """Track states and offer events for covers."""
+    component = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL, GROUP_NAME_ALL_COVERS)
+    component.setup(config)
+
+    def handle_cover_service(service):
+        """Handle calls to the cover services."""
+        method = SERVICE_TO_METHOD.get(service.service)
+        params = service.data.copy()
+        params.pop(ATTR_ENTITY_ID, None)
+
+        if method:
+            for cover in component.extract_from_service(service):
+                getattr(cover, method['method'])(**params)
+
+                if cover.should_poll:
+                    cover.update_ha_state(True)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    for service_name in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[service_name].get(
+            'schema', COVER_SERVICE_SCHEMA)
+        hass.services.register(DOMAIN, service_name, handle_cover_service,
+                               descriptions.get(service_name), schema=schema)
+    return True
+
+
+class CoverDevice(Entity):
+    """Representation a cover."""
+
+    # pylint: disable=no-self-use
+    @property
+    def current_cover_position(self):
+        """Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return None
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current position of cover tilt.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return None
+
+    @property
+    def state(self):
+        """Return the state of the cover."""
+        closed = self.is_closed
+
+        if closed is None:
+            return STATE_UNKNOWN
+
+        return STATE_CLOSED if closed else STATE_OPEN
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        data = {
+            ATTR_CURRENT_POSITION: self.current_cover_position
+        }
+
+        current_tilt = self.current_cover_tilt_position
+        if current_tilt is not None:
+            data[ATTR_CURRENT_TILT_POSITION] = self.current_cover_tilt_position
+
+        return data
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed or not."""
+        return None
```

Comment:
This should raise a `NotImplementedError()` as this is a mandatory property of a cover.